lib/core/fs/read_as_posix_file.dart

```diff
@@ -0,0 +1,23 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+/// Reads a [file] as bytes and replaces all DOS line feeds (`\r\n`)
+/// with Posix line feeds (`\n`).
+Future<Uint8List> readAsPosixFile(File file) async {
+  if (!await file.exists()) {
+    throw ArgumentError('The file at path "${file.path}" does not exist.');
+  }
+
+  final content = await file.readAsString();
+  final normalized = content.replaceAll('\r\n', '\n');
+
+  return Uint8List.fromList(utf8.encode(normalized));
+}
+
+/// Extension on [File] which provides a method to read it as Posix file.
+extension PosixReadable on File {
+  /// Reads the file as bytes and replaces all DOS line feeds (`\r\n`)
+  /// with Posix line feeds (`\n`).
+  Future<Uint8List> readAsPosixBytes() async => await readAsPosixFile(this);
```

Comment:
Why do they have different names?

Reply:
The function is named the way it is to make the purpose of reading a file clear. In the extension this is not required because we are already extending `File`. To comply with the naming schema of `File` (`readAsBytes`, `readAsString`, `readAsBytesSync`, ....) I've replaced File with Bytes.

Reply:
Sounds reasonable :)